### PR TITLE
UPSTREAM: Add AES-CBC and Secretbox encryption

### PR DESCRIPTION
**Pull request by smarterclayton**
Completes the pick of the upstream changes for Kube 1.7 for encryption at rest of secrets. Includes the new config loading.

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by mfojtik** [comment]
@php-coder FYI

**Comment by mfojtik** [comment]
flake: https://github.com/openshift/origin/issues/14496

[test]

**Comment by mfojtik** [comment]
@smarterclayton you need to tweak the pick for `vendor/golang.org/x/crypto` to pass the commit checker.

**Comment by mfojtik** [comment]
LGTM after fixing commigs //cc @php-coder for the config changes

**Comment by smarterclayton** [comment]
Of of Memory building network [test]

**Comment by smarterclayton** [comment]
Router flake, no route to host from integration TestRouter [test]

**Comment by smarterclayton** [comment]
[merge] (since this is pure upstream code)

I think we want to use the upstream config.

**Comment by smarterclayton** [comment]
Dind flake fedora:25 not found?  Merging

**Comment by kargakis** [comment]
> Dind flake fedora:25 not found? Merging

Yeah, seen that elsewhere and it seems to block the queue now

**Comment by kargakis** [comment]
Opened https://github.com/openshift/origin/issues/14573

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+	"k8s.io/apiserver/pkg/storage/value"
+	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/secretbox"
+)
+
+const (
+	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
+	aesGCMTransformerPrefixV1    = "k8s:enc:aesgcm:v1:"
+	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1"
```
Comment (php-coder) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go]: @smarterclayton I see that we haven't backported fix for this: #47537 Do we need it to be backported?

**Comment by smarterclayton** [comment]
Yup

On Jun 19, 2017, at 1:03 PM, Vyacheslav Semushin <notifications@github.com>
wrote:

*@php-coder* commented on this pull request.
------------------------------

In
vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go
<https://github.com/openshift/origin/pull/14517#discussion_r122765124>:

> +	"io/ioutil"
+	"os"
+
+	yaml "github.com/ghodss/yaml"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage/value"
+	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/secretbox"
+)
+
+const (
+	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
+	aesGCMTransformerPrefixV1    = "k8s:enc:aesgcm:v1:"
+	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1"

@smarterclayton <https://github.com/smarterclayton> I see that we haven't
backported fix for this: #47537 Do we need it to be backported?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/14517#pullrequestreview-44909310>,
or mute the thread

**Comment by php-coder** [comment]
@smarterclayton Ok, I'll create a PR for that.

**Comment by php-coder** [comment]
[[security] [SCCFSI] Encrypt secrets at REST in etcd](https://trello.com/c/yhuQZ3NP/905-security-sccfsi-encrypt-secrets-at-rest-in-etcd)

**Pull request merged**